os/net/app-layer/coap/coap-constants.h

```diff
@@ -145,29 +145,140 @@ typedef enum {
+  APPLICATION_SENSML_CBOR = 113,
+  /* application/senml-exi; Ref: [RFC8428] */
+  APPLICATION_SENML_EXI = 114,
+  /* application/sensml-exi; Ref: [RFC8428] */
+  APPLICATION_SENSML_EXI = 115,
+  /* application/yang-data+cbor; id=sid; Ref: [RFC9254] */
+  APPLICATION_YANG_DATA_CBOR__ID_SID = 140,
+  /* application/coap-group+json; Ref: [RFC7390] */
+  APPLICATION_COAP_GROUP_JSON = 256,
+  /* application/concise-problem-details+cbor; Ref: [RFC9290] */
+  APPLICATION_CONCISE_PROBLEM_DETAILS_CBOR = 257,
+  /* application/swid+cbor; Ref: [RFC9393] */
+  APPLICATION_SWID_CBOR = 258,
+  /* application/pkixcmp; Ref: [RFC-ietf-ace-cmpv2-coap-transport-10][RFC4210] */
+  APPLICATION_PKIXCMP = 259,
+  /* application/dots+cbor; Ref: [RFC9132] */
+  APPLICATION_DOTS_CBOR = 271,
+  /* application/missing-blocks+cbor-seq; Ref: [RFC9177] */
+  APPLICATION_MISSING_BLOCKS_CBOR_SEQ = 272,
+  /* application/pkcs7-mime; smime-type=server-generated-key; Ref: [RFC7030][RFC8551][RFC9148] */
+  APPLICATION_PKCS7_MIME__SMIME_TYPE_SERVER_GENERATED_KEY = 280,
+  /* application/pkcs7-mime; smime-type=certs-only; Ref: [RFC8551][RFC9148] */
+  APPLICATION_PKCS7_MIME__SMIME_TYPE_CERTS_ONLY = 281,
+  /* application/pkcs8; Ref: [RFC5958][RFC8551][RFC9148] */
+  APPLICATION_PKCS8 = 284,
+  /* application/csrattrs; Ref: [RFC7030][RFC9148] */
+  APPLICATION_CSRATTRS = 285,
+  /* application/pkcs10; Ref: [RFC5967][RFC8551][RFC9148] */
+  APPLICATION_PKCS10 = 286,
+  /* application/pkix-cert; Ref: [RFC2585][RFC9148] */
+  APPLICATION_PKIX_CERT = 287,
+  /* application/aif+cbor; Ref: [RFC9237] */
+  APPLICATION_AIF_CBOR = 290,
+  /* application/aif+json; Ref: [RFC9237] */
+  APPLICATION_AIF_JSON = 291,
+  /* application/senml+xml; Ref: [RFC8428] */
+  APPLICATION_SENML_XML = 310,
+  /* application/sensml+xml; Ref: [RFC8428] */
+  APPLICATION_SENSML_XML = 311,
+  /* application/senml-etch+json; Ref: [RFC8790] */
+  APPLICATION_SENML_ETCH_JSON = 320,
+  /* application/senml-etch+cbor; Ref: [RFC8790] */
+  APPLICATION_SENML_ETCH_CBOR = 322,
+  /* application/yang-data+cbor; Ref: [RFC9254] */
+  APPLICATION_YANG_DATA_CBOR = 340,
+  /* application/yang-data+cbor; id=name; Ref: [RFC9254] */
+  APPLICATION_YANG_DATA_CBOR__ID_NAME = 341,
+  /* application/td+json; Ref: ["Web of Things (WoT) Thing Description 1.1", April 2022] */
+  APPLICATION_TD_JSON = 432,
+  /* application/tm+json; Ref: ["Web of Things (WoT) Thing Description 1.1", April 2022] */
+  APPLICATION_TM_JSON = 433,
+  /* application/voucher-cose+cbor; Ref: [draft-ietf-anima-constrained-voucher-20] */
+  APPLICATION_VOUCHER_COSE_CBOR = 836,
+  /* application/vnd.ocf+cbor; Ref: [Michael_Koster] */
+  APPLICATION_VND_OCF_CBOR = 10000,
+  /* application/oscore; Ref: [RFC8613] */
+  APPLICATION_OSCORE = 10001,
+  /* application/javascript; Ref: [RFC4329] */
+  APPLICATION_JAVASCRIPT = 10002,
+  /* application/json; deflate; Ref: [RFC8259][RFC9110, Section 8.4.1.2] */
+  APPLICATION_JSON = 11050,
```

Comment:
I haven't looked at the script, but the CI logs show it does not generate different names for `application/json` and `application/json; deflate` and a few others.

Reply:
Yeah good catch, looks like content_coding needs to be taken into account as well not just content_type.

Just pushed my change to update the gen tool and regenerated the header